### DISCUSSION TITLE
Can't use $_ENV[] in php commands when switching from 2.x image to 3.x image

### DISCUSSION THREAD

**Discussion by mnavsa-openbox** [docksal/docksal#1701] [Q&A]
Are there any stable 2.x images of PHP8.0 available, or is 3.x the only option? 

EDIT: I updated the title to be a bit more specific regarding the issue I am investigating.

**Comment by sean-e-dietrich** [comment]
Hey there, @mnavsa-openbox there will not be any 2.x versions of 8.0, unfortunately. The main reason is that as part of 3.x we introduced architecture changes that don't really backport with those images.

**Reply by mnavsa-openbox** [reply]
Thank you for the quick response!

Currently I am working on upgrading from 2-php7.4 to php8.0. The site I am working on has custom Docksal commands which read environment variables like this: `$_ENV['DOCROOT']`. After initializing the site container with php8.0 the following error is received: `Warning: Undefined array key "DOCROOT"`. 

This issue is present on php7.4-3 containers as well.

Are you able to provide insight into what can be done to have these variables read correctly?

**Comment by shelane** [comment]
You are using php custom commands? Have you tried using `getenv('DOCROOT')`? I use bash custom commands and I know that the `DOCROOT` environment variable is set in the newest images.

**Reply by mnavsa-openbox** [reply]
Thanks for the response. Updating the commands to use getenv() instead of $_ENV() is working on my setup. However, I'm trying to determine if there is a way I can upgrade the CLI without changing any of the commands. Given that the issue is present on php7.4-3 image as well, I'd like to know if there are any container settings I can tweak to get the current implementation to work since this still is a valid way of reading environment variables in PHP.

**Comment by shelane** [comment]
I did a search on PHP and this comment seemed interesting:

>If your $_ENV array is mysteriously empty, but you still see the variables when calling getenv() or in your phpinfo(), check your http://us.php.net/manual/en/ini.core.php#ini.variables-order ini setting to ensure it includes "E" in the string.

We switched to use getenv() years ago because it was not dependent on that setting.

**Reply by mnavsa-openbox** [reply]
This works :)

Any reason why this changed between 2.x and 3.x?
